### PR TITLE
Item Appearances for things with multiple classifications

### PR DESCRIPTION
**Pull request by ScipioWright**
Things can be multiple item classifications at once, so I think we should display that in some fashion.
Currently, just a proof of concept mostly, showing that it does, in fact, function properly.

Things we should have appearances for:
- Progression + Useful (to indicate the cool important items, like hookshot or sword)
- Progression + Trap (very niche, would apply to Hard Mode in Terraria or potentially the Dreamers in Hollow Knight)
- Maybe Trap + Useful? Not sure what that would even be, but it's probably not difficult to come up with something generic for it.
- Progression + Useful + Trap? idk why someone would do this but might as well

Of note, filler is just the absence of the other flags, so nothing can be filler + something else.

### CODE REVIEW

**Comment by heinermann** [comment]
I don't see value in overcomplicating it. We can just leave it as is if it's a trap. Also should be an easier way to do that code without nesting ifs.

**Comment by ScipioWright** [comment]
For progression + trap, we definitely don't want it to just appear as a trap

File: data/archipelago/scripts/ap_utils.lua [hunk 1]
```diff
@@ -258,12 +258,20 @@ function create_ap_entity_from_flags(location, x, y)
...
+		if bit.band(flags, AP.ITEM_FLAG_USEFUL) ~= 0 then
+			item_filename = "ap_progression_useful_shopitem.xml"
+			item_description = "$ap_shopdescription_progression_useful"
+		elseif bit.band(flags, AP.ITEM_FLAG_TRAP) ~= 0 then
+			item_filename = "ap_progression_trap_shopitem.xml"
+			item_description = "$ap_shopdescription_progression_trap"
+		else
+			item_filename = "ap_progression_shopitem.xml"
+			item_description = "$ap_shopdescription_progression"
+		end
```
Comment (heinermann) [inline data/archipelago/scripts/ap_utils.lua:271]: I guess this is fine. The alternative would be making new fields for the combinations and doing `if bit.band(flags, AP.ITEM_FLAG_USEFUL_PROGRESSION) == AP.ITEM_FLAG_USEFUL_PROGRESSION then` which is worse visually.

**Comment by ScipioWright** [comment]
Closing this to reopen with a different version

**Pull request closed without merging**